dashboard/src/shared/Auth.ts

```diff
@@ -90,6 +89,16 @@ export class Auth {
     }
   }
 
+  // is403FromAPIsServer returns true if the response is a 403 determined to have originated
+  // from the grpc-web APIs server, rather than the auth proxy.
+  public static is403FromAPIsServer(e: any): boolean {
+    const contentType = e.metadata?.headersMap["content-type"] as string[];
+    if (contentType.some(v => v === "application/grpc-web+proto")) {
```

Comment:
is it always the case that if the exception contains this content type is a 403?

Reply:
Just FTR, the grpc-gw fn to determine which is a grpcwerb request is: 

```go
func (w *WrappedGrpcServer) IsGrpcWebRequest(req *http.Request) bool {
	return req.Method == http.MethodPost && strings.HasPrefix(req.Header.Get("content-type"), "application/grpc-web")
}
```

so  perhaps, instead of `===` we could check the prefix to cover all the possible cases of the [grpc-web protocol](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-WEB.md#protocol-differences-vs-grpc-over-http2), namely:

- application/grpc-web+proto
- application/grpc-web+json
- application/grpc-web+thrift
- application/grpc-web-text+proto
- application/grpc-web-text+thrift

Reply:
It's not an assumption that if the exception contains this content type, that it is a 403 error. The function itself (`is403FromAPIsServer`) assumes that the argument *is* a 403 already, so is checking the content type of the given 403 error. Though it doesn't need to assume that, I suppose?

Maybe that's what you mean? We could (and perhaps should) rename this function as `isErrorFromAPIsServer()`, since the fact that it's currently being used to check 403 errors only isn't relevant to the function itself? I'll update to that.

Reply:
> so perhaps, instead of `===` we could check the prefix to cover all the possible cases of the [grpc-web protocol](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-WEB.md#protocol-differences-vs-grpc-over-http2), 

Yep, makes sense, will update to that. Thanks.

Reply:
Yeah, that was my question. Thanks for the name change!